recipes/fragpipe/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "FragPipe" %}
+{% set version = "20.0" %}
+{% set philosopher_version = "5.1.0" %}
+
+# These keys were generated ONLY for the testing of this bioconda package.
+# Users must generate their own key by agreeing to the terms at https://msfragger-upgrader.nesvilab.org/upgrader/ and https://msfragger.arsci.com/ionquant/.
+{% set msfragger_academic_use_only_key = "f28c4be2-759b0a11-6c1ea652-1240565a" %}
+{% set ionquant_academic_use_only_key = "47cb2953-b4bf7726-2c96f5fe-8f1013aa" %}
+
+package:
+  name: fragpipe
+  version: {{ version }}
+
+build:
+  number: 0
+  noarch: generic
+  run_exports:
+    - {{ pin_subpackage('fragpipe', max_pin="x.x") }}
+
+source:
+  - url: https://github.com/Nesvilab/{{ name }}/releases/download/{{ version }}/{{ name }}-{{ version }}.zip
+    sha256: 0d77db85ae26c190a915823e1c68238f3a19533810734b008564b2a8d85d1bd6
+    folder: fragpipe_source
+  - url: https://github.com/Nesvilab/philosopher/releases/download/v{{ philosopher_version }}/philosopher_v{{ philosopher_version }}_linux_amd64.zip
+    sha256: cf15a2d6321ecb530170b2410f5873f89e283b52d5f1ab0f27eebc124f4f4a1a
+    folder: philosopher_source
+
+requirements:
+  run:
+    - openjdk =11
+    - python =3.9
+    - msfragger =4.0
+    - ionquant =1.10.12
+
+test:
+  commands:
+    - fragpipe --msfragger_key {{ msfragger_academic_use_only_key }} --ionquant_key {{ ionquant_academic_use_only_key }} --help  2>&1 | grep "FragPipe v{{ version }}"
+
+about:
+  home: https://github.com/Nesvilab/FragPipe
+  license: GNU General Public License v3.0
```

Comment:
please use SPDX identifiers here

Reply:
It looks like since this recipe was started, the FragPipe LICENSE.txt now reflects that there are multiple components with their own licenses.

Most of these are permissive or GPL and I should be able to include the license files in this recipe, but there is one, DIA-NN, that may be more trouble: https://github.com/Nesvilab/FragPipe/blob/develop/MSFragger-GUI/tools/diann/DIA-NN%20FragPipe%20version%20license.txt:

> Transfer. You may not rent, lease, lend or sublicense the DIA-NN software. You may, however, make a one-time permanent transfer of all of your license rights to the DIA-NN software to another party, provided that..

Would you agree this poses a problem for the conda package? Worst-case scenario, this component is only used by a small number of workflows and could be left out for the time being.

Reply:
@bgruening - I added licenses for the all of the bundled software except for DIA-NN. This I've disabled as described above. The FragPipe developers did get special permission from the developer, which is not something we have for Bioconda at this time, to my knowledge.

Reply:
Thanks @reid-wagner for digging this out!